From 16583d0fbd1d0639d512e2852e111fd22ccfd8c6 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Sat, 22 Oct 2022 00:03:39 +0200
Subject: [PATCH 2/2] i2c: ljca: Call acpi_dev_clear_dependencies()

Call acpi_dev_clear_dependencies() to mark _DEP ACPI dependencies on
the I2C controller as satisfied so that acpi_dev_ready_for_enumeration()
for the I2C device nodes in APCI will return true once the I2C controller
is registered.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 drivers/i2c/busses/i2c-ljca.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-ljca.c b/drivers/i2c/busses/i2c-ljca.c
index b7dbb55..cd803bc 100644
--- a/drivers/i2c/busses/i2c-ljca.c
+++ b/drivers/i2c/busses/i2c-ljca.c
@@ -427,7 +427,14 @@ static int ljca_i2c_probe(struct platform_device *pdev)
 		return -EIO;
 	}
 
-	return i2c_add_adapter(&ljca_i2c->adap);
+	ret = i2c_add_adapter(&ljca_i2c->adap);
+	if (ret)
+		return ret;
+
+	if (has_acpi_companion(&ljca_i2c->adap.dev))
+		acpi_dev_clear_dependencies(ACPI_COMPANION(&ljca_i2c->adap.dev));
+
+	return 0;
 }
 
 static int ljca_i2c_remove(struct platform_device *pdev)
-- 
2.37.3

